File: Algorithms/Implementation/Migratory Birds.py
# Migratory Birds

# 1
def migratoryBirds(arr):
  # Write your code here
  res = [0] * 2
  hashMap = {}

  for i in arr:
    hashMap[i] = hashMap[i] + 1 if i in hashMap else 1

  for item in hashMap.items():
    val, count = item[0], item[1]

    if count > res[1] or count == res[1] and val < res[0]:
      res[0] = val
      res[1] = count

  return res[0]

File: Algorithms/Implementation/Migratory Birds.js
// Migratory Birds

// https://www.hackerrank.com/challenges/migratory-birds/problem

// #1
function migratoryBirds(arr) {
  // Write your code here
  const hashMap = {}
  const res = [0, 0]

  for (let el of arr) {
    hashMap[el] = (hashMap[el] || 0) + 1
  }

  for (const [val, count] of Object.entries(hashMap)) {
    if (count > res[1] || (count === res[1] && val < res[0])) {
      res[0] = +val
      res[1] = count
    }
  }

  return res[0]
}